server/userNotification/create.ts

```diff
@@ -0,0 +1,84 @@
+import { Op } from 'sequelize';
+
+import * as types from 'types';
+import { UserNotification, UserSubscription } from 'server/models';
+import { indexByProperty, splitArrayOn } from 'utils/arrays';
+import { filterUsersWhoCanSeeThread } from 'server/thread/queries';
+
+type ActivityItemResponder<Kind extends types.ActivityItemKind> = (
+	item: types.ActivityItemOfKind<Kind>,
+) => Promise<void>;
+
+const createNotificationsForThreadComment = async (
+	item: types.ActivityItemOfKind<'pub-discussion-comment-added' | 'pub-review-comment-added'>,
+	includePubLevelSubscribers: boolean,
+) => {
+	const {
+		actorId,
+		pubId,
+		payload: { threadId },
+	} = item;
+
+	const subscriptionWhereQueries = includePubLevelSubscribers
+		? [{ pubId }, { threadId }]
+		: [{ threadId }];
+
+	const subscriptions: types.UserSubscription[] = await UserSubscription.findAll({
+		where: {
+			[Op.or]: subscriptionWhereQueries,
+			userId: { [Op.not]: actorId },
+			status: { [Op.not]: 'inactive' },
+		},
+	});
+
+	const [mutedThreadSubscriptions, unmutedThreadSubscriptions] = splitArrayOn(
+		subscriptions.filter((sub) => !!sub.threadId),
+		(s) => s.status === 'muted',
+	);
+
+	const unmutedPubSubscriptionsNotSupersededByThreadMute = subscriptions.filter(
+		(sub) =>
+			sub.pubId &&
+			sub.status !== 'muted' &&
+			!mutedThreadSubscriptions.some((mutedSub) => mutedSub.userId === sub.userId),
+	);
+
+	const subscriptionsThatMayProduceNotifications = [
+		...unmutedThreadSubscriptions,
+		...unmutedPubSubscriptionsNotSupersededByThreadMute,
+	];
+
+	const subscriptionsByUserId = indexByProperty(
+		subscriptionsThatMayProduceNotifications,
+		'userId',
+	);
+
+	const userIdsToNotifty = await filterUsersWhoCanSeeThread({
```

Comment:
surely, typo for notify

Reply:
Fixed not here but in the `notifications` branch.